knotx-core/src/main/java/com/cognifide/knotx/repository/RemoteRepository.java

```diff
@@ -52,19 +54,14 @@ static RemoteRepository of(String path, String domain, Integer port, HttpClient
     }
 
     @Override
-    public void get(URI uri, AsyncResultHandler<Template<String, URI>> handler) throws IOException {
+    public void get(URI uri, MultiMap requestHeaders, AsyncResultHandler<Template<String, URI>> handler) throws IOException {
         HttpClientRequest httpClientRequest = httpClient.get(port, domain, uri.toString(),
-                httpClientResponse -> {
-                    if (httpClientResponse.statusCode() == HttpStatus.OK.value()) {
-                        httpClientResponse.bodyHandler(buffer -> {
-                            String responseContent = buffer.getString(0, buffer.length());
-                            handler.handle(AsyncResultFactory.createSuccess(new BasicTemplate(uri,
-                                    responseContent)));
-                        });
-                    } else {
-                        handler.handle(AsyncResultFactory.createFailure());
-                    }
-                });
+                httpClientResponse -> httpClientResponse.bodyHandler(buffer -> {
+                    String responseContent = buffer.getString(0, buffer.length());
+                    handler.handle(AsyncResultFactory.createSuccess(
+                            new HttpTemplate(uri, httpClientResponse.statusCode(), httpClientResponse.headers(), responseContent)));
+                }));
+        requestHeaders.forEach(stringStringEntry -> httpClientRequest.putHeader(stringStringEntry.getKey(), stringStringEntry.getValue()));
```

Comment:
Please change variable `stringStringEntry` name to more meaningful e.g. `headerEntry`.